TabMonConfigBuilder/TabMonConfigBuilder.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using TabMonConfigBuilder.Helpers;
+
+namespace TabMonConfigBuilder
+{
+    public sealed class TabMonConfigBuilder
+    {
+        private readonly CommandLineOptions commandLineOptions;
+        private readonly string currentWorkingDirectory;
+
+        public TabMonConfigBuilder(CommandLineOptions commandLineOptions, string currentWorkingDirectory)
+        {
+            this.commandLineOptions = commandLineOptions;
+            this.currentWorkingDirectory = currentWorkingDirectory;
+        }
+
+        #region Public Methods
+
+        /// <summary>
+        /// Sets up and issues the LogsharkRequest to the LogsharkController.
+        /// </summary>
+        public void Execute()
+        {
+            var hosts = new Dictionary<string, Host>();
+
+            Console.WriteLine("Parsing topology file for process entries..");
+            try
+            {
+                ParseTopologyAndUpdateHosts(commandLineOptions.Target, hosts);
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+
+            Console.WriteLine(string.Format("Writing config section and instructions to {0}..", commandLineOptions.Output));
+            try
+            {
+                WriteToFile(commandLineOptions.Output, hosts);
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+            Console.WriteLine("Done writing config section..");
+        }
+
+        #endregion Public Methods
+
+        #region Private Methods
+
+        private static void ParseTopologyAndUpdateHosts(string target, Dictionary<string, Host> hosts)
```

Comment:
Can we make this func return a Dictionary<string, Host> instead of taking an empty list as a parameter?